dom.bs

```diff
@@ -6835,9 +6887,15 @@ for <a>this</a>.
  <dd><p><a>Throw</a> a "{{SyntaxError!!exception}}" {{DOMException}}.
 </dl>
 
+<p>The
+<dfn method for=Element><code>getInnerHTML(<var>options</var>)</code></dfn>
+method, when invoked, must return the result of running <a>HTML fragment serialization algorithm</a>,
+given <a>this</a> as <var>node</var>, <i>options.includeShadowRoots</i> as <i>include shadow roots</i>,
+and <i>options.closedRoots</i> as <i>closed shadow roots</i>.
```

Comment:
If we are to add this feature, I suspect we want to define this in HTML instead, @domenic?

Reply:
Yeah, the plan is to move https://w3c.github.io/DOM-Parsing/#the-innerhtml-mixin into HTML, not DOM. One day...

Reply:
So just to be clear here - you'd prefer I move this definition (lines 6890-6894) to the HTML spec, but leave the IDL that defines `getInnerHTML` here in DOM. Is that right? Would you mind pointing me to another example that auto-magically links over to HTML from the IDL like this, so I can copy it? I looked through the [IDL Index](https://dom.spec.whatwg.org/#idl-index) but didn't find any such examples. I'm sure I missed them.

Reply:
No, we should also move the IDL. HTML will define all of `innerHTML`, including that.

Reply:
So as part of this PR (adding declarative Shadow DOM), you also want me to move all of the Element IDL out of here and into HTML? I suppose I can try doing that, but it seems like perhaps that should be a separate PR? This one is already fairly large, and is moving along pretty slowly. I would hate to slow it down further. Having said that, I'll do it if you think it's necessary here.

Reply:
My thought was that HTML would have `partial interface Element {}` to extend this interface, but having said that, that's also a thing we're considering to stop doing. Hmm. (We definitely don't want to move all of `Element`.)

Reply:
`partial interface Element {}` is what I anticipated too. I don't think we're going to stop using partial interfaces any time soon (for `Document` alone, that would basically require merging the HTML and DOM specs). That also matches how https://w3c.github.io/DOM-Parsing/#extensions-to-the-element-interface works.